src/main/java/com/google/api/codegen/transformer/JavaSurfaceNamer.java

```diff
@@ -0,0 +1,227 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.CollectionConfig;
+import com.google.api.codegen.MethodConfig;
+import com.google.api.codegen.ServiceMessages;
+import com.google.api.codegen.java.JavaDocUtil;
+import com.google.api.codegen.metacode.InitValueConfig;
+import com.google.api.codegen.util.Name;
+import com.google.api.tools.framework.aspects.documentation.model.DocumentationUtil;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.TypeRef;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class JavaSurfaceNamer implements SurfaceNamer {
```

Comment:
In this case I specifically chose "Surface" because the names in this interface are for the surface. For your test generation, you'll probably want your own namer.